.github/workflows/build-and-push-image-prod.yml

```diff
@@ -0,0 +1,58 @@
+name: Continuous Delivery / Prod
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  set-test-release-version:
+    name: Set Test Release Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{steps.release-version.outputs.version}}
+    steps:
+      - id: release-version
+        run: echo "::set-output name=version::release-7"
+
+  build-and-push-image-test:
+    name: Build and push image test
+
+    runs-on: ubuntu-latest
+    needs: set-test-release-version
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{needs.set-test-release-version.outputs.version}}
+
+      - name: Azure Container Registry login
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.PROD_AZURE_ACR_CLIENTID }}
+          password: ${{ secrets.PROD_AZURE_ACR_SECRET }}
+          registry: ${{ secrets.PROD_ACR_URL }}
+
+      - name: Prepare tags
+        id: prepare-tags
+        run: |
+          DOCKER_IMAGE=${{ secrets.PROD_ACR_URL }}/complete-app
+          VERSION=latest
+          TAGS="${DOCKER_IMAGE}:${VERSION}"
+          if [ "${{ github.event_name }}" = "push" ]; then
+            VERSION=sha-${GITHUB_SHA}
+            TAGS="$TAGS,${DOCKER_IMAGE}:${VERSION},${DOCKER_IMAGE}:${{needs.set-test-release-version.outputs.version}}"
+          fi
+          echo ::set-output name=tags::${TAGS}
+          echo ::set-output name=deploy-version::${VERSION}
+
+      - name: Push image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          build-args: |
```

Comment:
Why do we need `build-args` and why is the `RAILS_ENV` development? I've never understood.